__tests__/Formsy.spec.tsx

```diff
@@ -1,59 +1,63 @@
 /* eslint-disable max-classes-per-file, react/destructuring-assignment */
-import { mount } from 'enzyme';
+import { createEvent, fireEvent, render } from '@testing-library/react';
 import * as React from 'react';
-import { useState } from 'react';
+import { useRef, useState } from 'react';
 
 import DynamicInputForm from '../__test_utils__/DynamicInputForm';
-import { getFormInstance, getWrapperInstance } from '../__test_utils__/getInput';
 import TestInput from '../__test_utils__/TestInput';
 import TestInputHoc from '../__test_utils__/TestInputHoc';
 import Formsy, { addValidationRule } from '../src';
 import { ValidationError } from '../src/interfaces';
 
 describe('Setting up a form', () => {
   it('should expose the users DOM node through an innerRef prop', () => {
-    class TestForm extends React.Component {
-      public inputRef: any;
+    const refSpy = jest.fn();
 
+    class TestForm extends React.Component {
       render() {
         return (
           <Formsy>
             <TestInputHoc
               name="name"
-              innerRef={(c) => {
-                this.inputRef = c;
+              innerRef={(ref: any) => {
+                if (!ref) {
+                  return;
+                }
+
+                refSpy(ref.constructor.name);
               }}
+              testId="test-input"
             />
           </Formsy>
         );
       }
     }
 
-    const form = mount(<TestForm />);
-    const input = ((form.instance() as TestForm) as TestForm).inputRef;
-    expect(input.props.name).toEqual('name');
+    render(<TestForm />);
+
+    expect(refSpy).toHaveBeenCalledWith('TestComponent');
   });
 
   it('should render a form into the document', () => {
-    const form = mount(<Formsy />);
-    expect(form.find('form').name()).toEqual('form');
+    const screen = render(<Formsy data-testid="form" />);
+    const form = screen.getByTestId('form') as HTMLFormElement;
+
+    expect(form.tagName.toLowerCase()).toEqual('form');
```

Comment:
`getByTestId` actually throws an error if no element is found, so this may not be necessary, since by definition if it gets to this point, it's true.

Reply:
I do think that it is better to check that the element it self is a form :]